controllers/appgroup_controller.go

```diff
@@ -110,57 +118,40 @@ func (r *ApplicationGroupReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		controllerutil.AddFinalizer(appGroup, v1alpha1.AppGroupFinalizer)
 		if err := r.Patch(ctx, appGroup, patch); err != nil {
 			logr.Error(err, "failed to patch the release with the appgroup finalizer")
-			return statusHelper.Failed(ctx, appGroup, err)
+			return ctrl.Result{}, err
```

Comment:
The function either returns `result, error` or `ctrl.Result, error`. Can we stick to one pattern of either returning the named returns?

Reply:
Don't quite get this comment, can you expound on this?

Reply:
Across this method - I either see a return `result, err` (i.e. using the named return param) or an explicit return value. I was recommending always using the named return params as the return values

Reply:
Where else do we use the result keyword for named return?

Reply:
For instance we have this block
```go
	if !appGroup.DeletionTimestamp.IsZero() {
		statusHelper.MarkTerminating(appGroup)
		result, err := reconcileHelper.Reverse(ctx)
		if !result.Requeue && err == nil {
			// Remove the finalizer because we have finished reversing
			controllerutil.RemoveFinalizer(appGroup, v1alpha1.AppGroupFinalizer)
			if err := r.Patch(ctx, appGroup, patch); err != nil {
				return ctrl.Result{}, err
			}
		}
		return result, err
	}
	```

	Here in `RemoveFinalizer` error case we return `ctrl.Result{}` while 2 lines below we return the named param `result`.
	So instead of returning `ctrl.Result{}` why not just return `result` for the former as well?

Reply:
result is coming from the `Reverse` method, it isn't necessarily directly intended to be a named return parameter. I guess it doesn't really matter one way or another for the patch call, since we get into that state, we are already returning an `err`

Reply:
i.e 
```go
			if err := r.Patch(ctx, appGroup, patch); err != nil {
				return ctrl.Result{}, err
			}
```
is the same as,
```go
			if err := r.Patch(ctx, appGroup, patch); err != nil {
				return result, err
			}
```

Reply:
All of this is mostly nit-picking for better readability - none of this affects the code quality :). Where I am coming from is based on the change in the function signature where the return params were named in this review. Nevertheless all of this is minor and it works fine.

Reply:
It is and isn't. Since the `result` is coming from the result of the `reconcileHelper.Reverse()` function, it is also possible that `result = ctrl.Result{RequeueAfter:...}`

Reply:
Ah got it!!